blocktx/peer_handler.go

```diff
@@ -275,6 +276,20 @@ func (bs *PeerHandler) HandleBlock(wireMsg wire.Message, peer p2p.PeerI) error {
 		gocore.NewStat("blocktx").NewStat("HandleBlock").AddTime(start)
 	}()
 
+	primaryBlocktx, err := bs.store.PrimaryBlocktx(context.Background())
+	if err != nil {
+		return err
+	}
+
```

Comment:
The checking if the `blocktx` is primary should happen earlier I think in `HandleBlockAnnouncement`. This would then avoid that a block is even requested by this `utils.SafeSend(bs.workerCh, pair)`

Reply:
Yes but `HandleBlock` is called "from outside" right? So we need to have check there. No?

Reply:
I think `HandleBlock` will not be called for a particular when `blocktx` does not request the block. But yes, let's have it in both places.